### ISSUE TITLE
Tracer Logs not showing when "logger" config field is provided

### ISSUE THREAD

**Issue by GeoffreyPlitt** [DataDog/dd-trace-js#3168]
We're using dd-trace to send APM logs to DataDog from NodeJS/Hapi, and it works well.

However, I am now debugging an issue with DataDog APM tracing (runMetrics not sent even when runMetrics:true is set in init), and DataDog support has asked for "Tracer Debug logs". So I've added a "logger" field to our init to enable tracer logs, per https://docs.datadoghq.com/tracing/troubleshooting/tracer_debug_logs/?code-lang=nodejs#review-debug-logs.

Here's our tracer init:

tracer.init({
		runtimeMetrics: true,
		logger: {
			debug: console.log,
			error: console.log,
			warn: console.log,
			info: console.log,
		},
	});

However, when we run our server and make requests, there are no tracer logs printed. Everything else looks fine but there are no tracer logs printing. Is this a bug? Are the docs wrong?

Node 14.21.3
dd-trace 3.21.0

**Comment by anmol242** [comment]
hey @GeoffreyPlitt , can you try by explicitly setting the log level to `debug`?

```
const tracer = require('dd-trace').init({
  runtimeMetrics: true,
  logger: {
    logLevel: 'debug', 
    debug: console.log,
    error: console.log,
    warn: console.log,
    info: console.log,
  },
});
```

**Comment by GeoffreyPlitt** [comment]
Yes, the env var is true, and I tried logLevel:'debug' but still don't see tracer logs. Perhaps your suggestion is for tracer v4? I cannot upgrade to 4 due to Node version constraint. In fact, I get typescript warnings that "logLevel" is not an allowed option.

(dd-trace 3.21.0)
